service/evidence/evidence_store_test.go

```diff
@@ -293,34 +295,205 @@ func TestService_StoreEvidences(t *testing.T) {
+					PageSize:  evidencetest.MockListEvidenceRequest2.PageSize,
+					PageToken: evidencetest.MockListEvidenceRequest2.PageToken,
+					OrderBy:   evidencetest.MockListEvidenceRequest2.OrderBy,
+					Asc:       evidencetest.MockListEvidenceRequest2.Asc,
+					Filter: &evidence.Filter{
+						ToolId: evidencetest.MockListEvidenceRequest2.Filter.ToolId,
+					},
+				},
+			},
+			wantErr: assert.NoError,
+			wantResp: func(tt assert.TestingT, i1 interface{}, i2 ...interface{}) bool {
+				res, ok := i1.(*evidence.ListEvidencesResponse)
+				assert.True(t, ok)
+
+				assert.Equal(t, 1, len(res.Evidences))
+
+				// Loop through all received evidences and check whether tool and service ids are correct.
+				for _, r := range res.Evidences {
+					assert.Equal(t, *evidencetest.MockListEvidenceRequest2.Filter.CloudServiceId, r.CloudServiceId)
+					assert.Equal(t, *evidencetest.MockListEvidenceRequest2.Filter.ToolId, r.ToolId)
+				}
+
+				return true
+			},
+		},
+		{
+			name: "Wrong Input handled correctly (req = nil)",
+			args: args{
+				in0: context.TODO(),
+				req: nil,
+			},
+			wantErr:  assert.Error,
+			wantResp: assert.Nil,
+		},
+		{
+			name: "Wrong Input handled correctly (tool_id not UUID)",
+			args: args{
+				in0: nil,
+				req: &evidence.ListEvidencesRequest{
+					PageSize:  evidencetest.MockListEvidenceRequest2.PageSize,
+					PageToken: evidencetest.MockListEvidenceRequest2.PageToken,
+					OrderBy:   evidencetest.MockListEvidenceRequest2.OrderBy,
+					Asc:       evidencetest.MockListEvidenceRequest2.Asc,
+					Filter: &evidence.Filter{
+						ToolId: util.Ref("No UUID Fromat"),
+					},
+				},
+			},
+			wantErr:  assert.Error,
+			wantResp: assert.Nil,
+		},
+		{
+			name: "Wrong Input handled correctly (cloud_service_id not UUID)",
+			args: args{
+				in0: nil,
+				req: &evidence.ListEvidencesRequest{
+					PageSize:  evidencetest.MockListEvidenceRequest2.PageSize,
+					PageToken: evidencetest.MockListEvidenceRequest2.PageToken,
+					OrderBy:   evidencetest.MockListEvidenceRequest2.OrderBy,
+					Asc:       evidencetest.MockListEvidenceRequest2.Asc,
+					Filter: &evidence.Filter{
+						CloudServiceId: util.Ref("No UUID Fromat"),
+					},
+				},
+			},
+			wantErr:  assert.Error,
+			wantResp: assert.Nil,
+		},
+		{
+			name: "Handle pagination error correctly",
+			fields: fields{
+				authz: &service.AuthorizationStrategyAllowAll{},
+				storage: testutil.NewInMemoryStorage(t, func(s persistence.Storage) {
+					assert.NoError(t, s.Create(&orchestratortest.MockAssessmentResult1))
+				}),
+			},
+			args: args{
+				in0: context.TODO(),
+				req: &evidence.ListEvidencesRequest{
+					PageSize:  evidencetest.MockListEvidenceRequest2.PageSize,
+					PageToken: evidencetest.MockListEvidenceRequest2.PageToken,
+					OrderBy:   "Wrong Input",
+					Asc:       evidencetest.MockListEvidenceRequest2.Asc,
+				},
+			},
+			wantErr:  assert.Error,
+			wantResp: assert.Nil,
+		},
+	}
+	for _, currentTest := range tests {
```

Comment:
Use `tt` please to be consistent with other code in this project. If we decide to change it later, we will do a full refactoring then.